functests/ptp/ptp.go

```diff
@@ -108,4 +97,472 @@ var _ = Describe("ptp", func() {
+						timeDiff, _ := execCommand(runningPod.Items[podNumber], collectTimeCommand)
+						offsetOK = compareOffsetTime(timeDiff)
+					}
+				}
+				Expect(offsetOK).ToNot(BeFalse(), "No salve pods detected")
+			})
+		})
+	})
+	var _ = Describe("Test NodeName Selector", func() {
+		var testPtpPod v1core.Pod
+		var ptpConfigSlave ptpv1.PtpConfig
+		path := "/apis/ptp.openshift.io/v1/namespaces/openshift-ptp/ptpconfigs"
+
+		BeforeEach(func() {
+			ptpConfigName := "test"
+
+			responce, err := client.Client.ConfigV1Interface.RESTClient().Get().AbsPath(path + "/slave").DoRaw()
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(responce, &ptpConfigSlave)
+			Expect(err).NotTo(HaveOccurred())
+			nodes, err := client.Client.Nodes().List(metav1.ListOptions{
+				LabelSelector: ptpSlaveNodeLabel,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(nodes.Items)).To(BeNumerically(">", 0),
+				fmt.Sprint("PTP Nodes with lable %s are not deployed on cluster", ptpSlaveNodeLabel))
+
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(ptpConfigSlave.GetSelfLink()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			ptpConfigTest := mutateProfile(ptpConfigSlave, ptpConfigName, nodes.Items[0].Name)
+
+			status = client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(path).
+				Resource("ptpconfigs").Body(ptpConfigTest).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP config creation Error"))
+
+			testPtpPod, err = getPtpPodOnNode(nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			testPtpPod, err = replaceTestPod(testPtpPod, 60)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		var _ = Context("Check if Node has Profile", func() {
+			//25743
+			It("PTP daemon can apply match rule based on nodeLabel", func() {
+				err := waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: test")
+				Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+			})
+		})
+		AfterEach(func() {
+			status := client.Client.ConfigV1Interface.RESTClient().Delete().AbsPath(path + "/test").Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("Can not delete PTP config"))
+
+			ptpConfigSlave.ObjectMeta.SetResourceVersion("")
+			status = client.Client.ConfigV1Interface.RESTClient().Post().AbsPath(path).
+				Resource("ptpconfigs").Body(ptpConfigSlave.DeepCopy()).Context(context.TODO()).Do()
+			Expect(status.Error()).NotTo(HaveOccurred(), fmt.Sprint("PTP slave config recovery Error"))
+			err := waitUntilLogIsDetected(testPtpPod, 180, "Profile Name: slave")
+			Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+		})
+	})
 })
+
+func podRole(runningPod v1core.Pod, role string) bool {
+	nodeList, err := client.Client.Nodes().List(metav1.ListOptions{
+		LabelSelector: role,
+	})
+	Expect(err).NotTo(HaveOccurred())
+	for NodeNumber := range nodeList.Items {
```

Comment:
nit: you can use the `ContainElement` instead of the for here

Reply:
Not sure if it's possible. I use methods of the nodeList structure